### PR TITLE
Fix (almost) all test failures

### PR DESCRIPTION
**Pull request by pgan002**
Fixed lots of bugs. The only failing test case is now one of implicit rage end year which is the following year after the range start. (https://iamplus.atlassian.net/browse/NC-344) which I will implement later.

### CODE REVIEW

**Review by scayetanot — APPROVED** [review]
looks good to me

File: tests/tests.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def test_this(self):
     def test_dow(self):
         now = datetime.now()
         hour_in_seconds = 24 * 60 * 60
-        for x, day in enumerate(('monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday')):
+        for x, day in enumerate(('monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'Saturday', 'sunday')):
```
Comment (brandonpulley) [inline tests/tests.py]: any reason the we are capitalizing just Saturday?

**Review by brandonpulley — APPROVED** [review]
looks good to me as long as the tests are passing locally

File: tests/tests.py (hunk 1 above)
Comment (pleasantone) [reply tests/tests.py]: Iirc,  a leftover typo from Steve.

**Pull request merged**